modules/platforms/cpp/protocol/include/ignite/protocol/reader.h

```diff
@@ -87,11 +89,46 @@ class Reader {
     [[nodiscard]] std::optional<std::string> readStringNullable();
 
     /**
-     * Read GUID.
+     * Read UUID.
      *
-     * @return GUID value.
+     * @return UUID value.
      */
-    [[nodiscard]] Guid readGuid();
+    [[nodiscard]] uuid readUuid();
+
+    /**
+     * Read Map size.
+     *
+     * @return Map size.
+     */
+    [[nodiscard]] uint32_t readMapSize() const {
+        checkDataInStream();
+
+        if (m_currentVal.data.type != MSGPACK_OBJECT_MAP)
+            throw ignite_error("The value in stream is not a Map");
+
+        return m_currentVal.data.via.map.size;
+    }
+
+    /**
+     * Read Map.
+     *
+     * @tparam K Key type.
+     * @tparam V Value type.
+     * @param handler Pair handler.
+     */
+     template<typename K, typename V>
+     void readMap(const std::function<void(K&&, V&&)>& handler) {
+        checkDataInStream();
+
+        if (m_currentVal.data.type != MSGPACK_OBJECT_MAP)
+            throw ignite_error("The value in stream is not a Map");
+
```

Comment:
Looks like  `readMapSize()` can be used here.

Reply:
Agree. Fixed